### ISSUE TITLE
Using `value_gen` in gd.py externally

### ISSUE THREAD

**Issue by zkytony** [pronobis/libspn#69]
I'm now working on an SPN model that uses discriminative GD learning. I would like to compute the loss on the training dataset as well as on the testing dataset, so as to see if the model overfits. I noticed that in gd.py, there are functions `cross_entropy_loss` and `mle_loss`, which creates internally an operation called `value_gen`. This object is exactly what would be helpful for my situation and I wonder if you could make this object accessible externally? Right now I have to modify the code in gd,py myself which is not a good thing to do.
